### DISCUSSION TITLE
RTL-SDR not working on RaspberryPi 4B

### DISCUSSION THREAD

**Discussion by pmaine** [AlexandreRouma/SDRPlusPlus#250] [Q&A] [answered]
SDRpp works fine on my Raspberry Pi 4B with my Airspy R2 and Spyverter. My RTL-SDR is not working. Please see the attachment. Any ideas?
![RTL-SDRandSDRpp](https://user-images.githubusercontent.com/5252230/129409722-161fb419-d742-4ddf-9a12-2328c75097f5.png)

After I hit the play button I see the following in the terminal window:

[2021-08-13 14:48:03.332] [error] No device selected
[2021-08-13 14:48:03.332] [info] RTLSDRSourceModule 'RTL-SDR Source': Tune: 133215490.6625!

I ran the following:

pi@hampi:~ $ rtl_test
Found 1 device(s):
  0:  Realtek, RTL2838UHIDIR, SN: 00000001

Using device 0: Generic RTL2832U OEM

Kernel driver is active, or device is claimed by second instance of librtlsdr.
In the first case, please either detach or blacklist the kernel module
(dvb_usb_rtl28xxu), or enable automatic detaching at compile time.

usb_claim_interface error -6
Failed to open rtlsdr device #0.

**Comment by the-gobleo-tinkerer** [comment]
Could you check if you've installed the rtl-sdr package? The error seems to say that the driver being used isn't the SDR one but the "stock" one intended for TV reception.

**Reply by pmaine** [reply]
Here are the commands that I have ran:

pi@hampi:~ $ sudo apt-get install rtl-sdr
Reading package lists... Done
Building dependency tree       
Reading state information... Done
rtl-sdr is already the newest version (0.6-1+rpt1).
0 upgraded, 0 newly installed, 0 to remove and 22 not upgraded.

pi@hampi:~ $ rtl_test
Found 1 device(s):
  0:  Realtek, RTL2838UHIDIR, SN: 00000001

Using device 0: Generic RTL2832U OEM

Kernel driver is active, or device is claimed by second instance of librtlsdr.
In the first case, please either detach or blacklist the kernel module
(dvb_usb_rtl28xxu), or enable automatic detaching at compile time.

usb_claim_interface error -6
Failed to open rtlsdr device #0.

**Reply by AlexandreRouma** [reply]
As @shuyuan-liu said, you didn't install the RTL-SDR drivers properly, make sure you've blacklisted the original rtl drivers.

**Reply by pmaine** [reply]
Can you guide me on how to blacklist the original drivers?

**Reply by the-gobleo-tinkerer** [reply]
Try rebooting the Pi? The config files to blacklist the DVB-T driver should have been installed in `/etc/udev/rules.d` by the rtl-sdr package, but I suspect it might need a reboot to take effect. Also could you check if there's a file named `rtlsdr.rules` or similar in the directory above?

**Reply by pmaine** [reply]
I was able to blacklist the drives using the process as follows:
sudo nano /etc/modprobe.d/blacklist-rtl.conf
Next I added the following three lines to the file:
blacklist dvb_usb_rtl28xxu
blacklist rtl2832
blacklist rtl2830

I next saved the file and exited the editor.

I next rebooted  using: shutdown -r now

After the system booted, I started SDRpp and it worked like a charm with  my RTL-SDR.

SDRpp is hands down the very best Linux SDR! I tip my hat to the developers! Next I need to send some financial support to them for their wonderful job!!!

Thanks again to everyone that is providing excellent help on this software!

**Reply by the-gobleo-tinkerer** [reply]
Glad it's working now! Would you mind marking the question as "answered"?

**Reply by pmaine** [reply]
Git won't allow me to mark as answered.

**Comment by pmaine** [comment, marked as answer]
See the above comments for the answer.
